### ISSUE TITLE
False-positive `N802` with `overload` and `override`

### ISSUE THREAD

**Issue by bersbersbers** [astral-sh/ruff#7479]
* A minimal code snippet that reproduces the bug.

```python
from typing import overload

from typing_extensions import override


class CannotModify:
    @overload
    def BadName(self, var: int) -> int:  # noqa: N802
        ...

    @overload
    def BadName(self, var: str) -> str:  # noqa: N802
        ...

    def BadName(self, var: int | str) -> int | str:  # noqa: N802
        return var


class MyClass(CannotModify):
    @overload
    def BadName(self, var: int) -> int:
        ...

    @overload
    def BadName(self, var: str) -> str:
        ...

    @override
    def BadName(self, var: int | str) -> int | str:
        return var * 2

```

* The command you invoked
```
ruff --isolated --select N802 bug.py
```

gives 

```
bug.py:21:9: N802 Function name `BadName` should be lowercase
bug.py:25:9: N802 Function name `BadName` should be lowercase
Found 2 errors.
```

I feel the `@override` before the implementation of `MyClass.BadName` should extend to the overloads. Or should I add one `@override` per `@overload`?

* The current Ruff version (`ruff --version`).

0.0.290

**Comment by charliermarsh** [comment]
I suppose we could just omit any methods tagged with `@overload` since presumedly we will end up flagging the implementation itself?

**Comment by charliermarsh** [comment]
Yeah, let's just omit `@overload` methods here I think.
